core/domain/user_query_services.py

```diff
@@ -38,3 +40,30 @@ def save_new_query_model(
         query_status=feconf.USER_QUERY_STATUS_PROCESSING,
         user_ids=[]).put()
     return query_id
+
+
+def send_email_to_qualified_users(
+        query_id, email_subject, email_body, email_intent, email_option,
+        num_of_users_to_send):
+    """Send email to maximum 'num_of_users_to_send' qualified users."""
+    query_model = user_models.UserQueryModel.get(query_id)
+    query_model.query_status = feconf.USER_QUERY_STATUS_ARCHIVED
+
+    if email_option == feconf.USER_QUERY_EMAIL_OPTION_ALL:
```

Comment:
Perhaps you can signal this case by having num_of_users_to_send be None. (Unless None already indicates something else, like "not yet populated"?)